registry/github_package_registry.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"log"
+	"strings"
+
+	"github.com/google/go-github/github"
+)
+
+// GithubPackageRegistry implements the Registry interface that talks to github and
+// expects packages in helm format without versioning and no qualifier in the path.
+// Format of the directory for a type is like so:
+// package/
+//   Chart.yaml
+//   manifests/
+//     foo.yaml
+//     bar.yaml
+//     ...
+type GithubPackageRegistry struct {
+	owner      string
+	repository string
+	client     *github.Client
+}
+
+// NewGithubRegistry creates a Registry that can be used to talk to github.
+func NewGithubPackageRegistry(owner, repository string) *GithubPackageRegistry {
+	return &GithubPackageRegistry{
+		owner:      owner,
+		repository: repository,
+		client:     github.NewClient(nil),
+	}
+}
+
+// List the types from the Registry.
+func (g *GithubPackageRegistry) List() ([]Type, error) {
+	// Just list all the types at the top level.
+	types, err := g.getDirs("")
```

Comment:
For now, I'm going to leave it as is due to github ratelimits. With 30 packages, this will not work. I'll add a TODO there that I had omitted from previous checkin once it became apparent this would be a crappy solution at the moment.